contracts/InterestRateModels/JumpRateModel.sol

```diff
@@ -88,23 +112,17 @@ contract JumpRateModel is InterestRateModel {
         }
     }
 
-    /**
-     * @notice Calculates the current supply rate per block
-     * @param cash The amount of cash in the market
-     * @param borrows The amount of borrows in the market
-     * @param reserves The amount of reserves in the market
-     * @param reserveFactorMantissa The current reserve factor for the market
-     * @return The supply rate percentage per block as a mantissa (scaled by 1e18)
-     */
-    function getSupplyRate(
-        uint cash,
-        uint borrows,
-        uint reserves,
-        uint reserveFactorMantissa
-    ) public view returns (uint) {
-        uint oneMinusReserveFactor = uint(1e18).sub(reserveFactorMantissa);
-        uint borrowRate = getBorrowRate(cash, borrows, reserves);
-        uint rateToPool = borrowRate.mul(oneMinusReserveFactor).div(1e18);
-        return utilizationRate(cash, borrows, reserves).mul(rateToPool).div(1e18);
+    function updateJumpRateModelInternal(
+        uint256 baseRatePerYear,
+        uint256 multiplierPerYear,
+        uint256 jumpMultiplierPerYear,
+        uint256 kink_
+    ) internal {
+        baseRatePerBlock = baseRatePerYear.div(blocksPerYear);
```

Comment:
Should we validate the input parameters here as well? Passing a very low value for `baseRatePerYear` might lead to precision loss and wrong calculation?

Reply:
Added require checks for that please check in this [commit](https://github.com/VenusProtocol/venus-protocol/pull/210/commits/46f0cf168db69632d3f6a108f589965f5544c579)